flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -313,13 +320,21 @@ public Builder uidPrefix(String newPrefix) {
 
       switch (writeMode) {
         case NONE:
-          return input;
+          if (!equalityFieldIds.isEmpty()) {
+            return input.keyBy(new EqualityFieldKeySelector(equalityFieldIds, table.schema(), flinkRowType));
+          } else {
+            return input;
+          }
 
         case HASH:
-          if (partitionSpec.isUnpartitioned()) {
-            return input;
-          } else {
+          if (!partitionSpec.isUnpartitioned() && !equalityFieldIds.isEmpty()) {
+            return input.keyBy(new HybridKeySelector(partitionSpec, equalityFieldIds, iSchema, flinkRowType));
+          } else if (!partitionSpec.isUnpartitioned() && equalityFieldIds.isEmpty()) {
             return input.keyBy(new PartitionKeySelector(partitionSpec, iSchema, flinkRowType));
+          } else if (partitionSpec.isUnpartitioned() && !equalityFieldIds.isEmpty()) {
+            return input.keyBy(new EqualityFieldKeySelector(equalityFieldIds, table.schema(), flinkRowType));
+          } else {
+            return input;
```

Comment:
It seems like we've gotten into a lot of `if` nesting. Would it make sense to either add a utility function, or add some method that returns more reader-friendly value from the equalityFieldIds? For me, its not as clear to see `isEmpty()` vs for example seeing `isUnpartitioned()`.

Can we add a utility or a method with some name that helps developers more easily understand the different cases, especially given we have so much `if else` nesting?

Reply:
That is make sense, I will add an utility method to do this.